docs/deployment/overview.md

```diff
@@ -1,47 +1,295 @@
+        job.Annotations.Add(new DeployingCallbackAnnotation(async ctx =>
+        {
+            CancellationToken ct = ctx.CancellationToken;
+
+            // Prompt the user for a confirmation using the interaction service
+            var interactionService = ctx.Services.GetRequiredService<IInteractionService>();
+
+            var envResult = await interactionService.PromptInputAsync(
+                "Environment Configuration",
+                "Please enter the target environment name:",
+                new InteractionInput
+                {
+                    Label = "Environment Name",
+                    InputType = InputType.Text,
+                    Required = true,
+                    Placeholder = "dev, staging, prod"
+                },
+                cancellationToken: ct);
+
+            // Custom deployment logic here
+            var reporter = ctx.ActivityReporter;
+            await using (var deployStep = await reporter.CreateStepAsync(
+                $"Deploying data seed job to {envResult.Value}", ct))
+            {
+                // Simulate deployment work
+                await Task.Delay(2000, ct);
+                await deployStep.SucceedAsync("Data seed job deployed successfully", ct);
+            }
+        }));
+
+        return resourceBuilder;
+    }
+}
+```
+
+This custom deployment logic integrates seamlessly with the `aspire deploy` command, providing interactive prompts and progress reporting.
+
+## Diagnostics & auditing
+
+Publishing gives you an immutable snapshot of intended structure before secrets appear. You can:
+
+- Diff published outputs between commits.
+- Scan for disallowed images or configuration.
+- Preserve a record for compliance, then separately record the resolved set applied at deployment time.
+
+## Additional tools
+
+### Azure Developer CLI (`azd`)
+
+[Azure Developer CLI (azd)](https://learn.microsoft.com/azure/developer/azure-developer-cli/) has first-class support for deploying Aspire projects. It can provision infrastructure, manage environments, and coordinate secret/value injection. You can incorporate Aspire publish artifacts into `azd` workflows or use the Azure integration (preview) directly.
+
+### Aspir8 (Kubernetes YAML generation)
+
+<a href="https://prom3theu5.github.io/aspirational">Aspir8 (Aspirate)</a> is a community tool that can transform an Aspire application model into Kubernetes manifests. While the Kubernetes hosting integration covers publish generation, Aspir8 may offer additional transforms or workflow preferences.
+
+## Legacy deployment manifest (footnote)
+
+Earlier workflows emphasized a single "deployment manifest" generated from specialized AppHost targets. The modern approach centers on `aspire publish` + integration extensibility. The legacy manifest format is **not being evolved further**, but you can still generate it for inspection or debugging:
+
+```bash
+aspire publish --publisher manifest -o diagnostics/
```

Comment:
This isn't how it was originally done, it was with a `dotnet` command...does this work?